File: hw/caiza/u3/hw29Strategy/SortingStrategySystem/src/main/java/ec/edu/espe/sortingstrategysystem/controller/InsertionSort.java
package ec.edu.espe.sortingstrategysystem.controller;
import ec.edu.espe.sortingstrategysystem.model.SortingStrategy;



/**
 *
 * @author Miguel Caiza,Overnight Developers Squad,DCCO-ESPE
 */
public class InsertionSort implements SortingStrategy{

    @Override
    public void sort(int[] array) {
        for (int i = 1; i < array.length; i++) {
            int key = array[i];
            int j = i - 1;
            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j = j - 1;
            }
            array[j + 1] = key;
        }
    }
    
}

File: exams/u3/saraguro/EXAMEN_3P_Strategy_Pattern/SortingApp/src/strategies/InsertionSort.js
class InsertionSort {
    sort(array) {
        let len = array.length;
        for (let i = 1; i < len; i++) {
            let key = array[i];
            let j = i - 1;
            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j = j - 1;
            }
            array[j + 1] = key;
        }
        return array;
    }
}

module.exports = InsertionSort;
